aspnet/fundamentals/logging/sample/src/LoggingSample/RequestLoggerMiddleware.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+using Microsoft.Framework.Logging;
+using Microsoft.Framework.Logging.Internal;
+
+namespace LoggingSample
+{
+    public class RequestLoggerMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger _logger;
+        private readonly static Random _random = new Random();
+
+        public RequestLoggerMiddleware(RequestDelegate next, 
+            ILoggerFactory loggerFactory)
+        {
+            _next = next;
+            _logger = loggerFactory.CreateLogger<RequestLoggerMiddleware>();
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            var eventId = _random.Next();
+            _logger.LogInformation(eventId, 
+                "{eventId}: Handling request: {path}", 
+                eventId, context.Request.Path);
+            try
+            {
+                await _next.Invoke(context);
+            }
+            catch (Exception ex)
+            {
+                if (_logger.IsEnabled(LogLevel.Critical))
+                {
+                    // perform expensive operation related to logging
+
+                    var values = new FormattedLogValues("{eventId}: Unexpected error handling request {path}:",
```

Comment:
Though, by switching to use my own pre-formatted message for LogCritical, aren't we giving up the ability for consumers of the log to search for the {Named} format parameters? Is there a way to keep that support, and log the exception, without using FormattedLogValues?